images/node/Dockerfile-debian

```diff
@@ -0,0 +1,51 @@
+###############################################################
+# DOCKER IMAGE - Debian based Nodejs Application (Express.js) #
+###############################################################
+
+FROM debian:latest
+
+# Install any dependencies for building & commands to use
+RUN apt-get update && apt-get install -y python3 make curl
+
+# Set verions of using packages
+ARG NODE_VERSION=18
+
+# Install NodeJS
+RUN apt-get update -yq \
+    && apt-get install -yq ca-certificates curl gnupg \
+    && mkdir -p /etc/apt/keyrings \
+    && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
+    && echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_VERSION.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
+    && apt-get update -yq \
+    && apt-get install nodejs -yq \
+    && npm install -g npm
+
+# Set the working directory within the container
+WORKDIR /app
+
+# Copy both package.json and package-lock.json
+COPY package*.json ./
+
+# Copy the application code into the container
+COPY . /app
+
+# Install application dependencies
+RUN npm install
```

Comment:
might use this instead: 

    npm ci --omit=dev

Reply:
Also lets move the "npm install" step to be above the "COPY . /app" step.

Reply:
That way docker caching works better.

Reply:
Lets also add a sample .dockerignore that ignores, by default, node_modules/ and .git/